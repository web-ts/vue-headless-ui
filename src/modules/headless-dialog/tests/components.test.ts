import { afterEach, describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import { HDialog, HDialogContent, HDialogTitle, HDialogDescription } from "../";

document.body.innerHTML = '<div id="app"></div>';

const dialog = defineComponent({
  props: {
    openByDefault: { type: Boolean, default: false }
  },
  setup(props) {
    const open = ref(props.openByDefault);

    function updateOpen(value: boolean) {
      open.value = value;
    }

    return () => [
      h("div", {}, h("button", { id: "open", onClick: () => updateOpen(!open.value) }, "Open Dialog")),
      h(HDialog, { id: "fixed_id", modelValue: open.value, "onUpdate:modelValue": (value) => updateOpen(value) }, () =>
        h(HDialogContent, {}, () => [
          h(HDialogTitle, { htmlTag: "h1" }, () => "dialog_title"),
          h(HDialogDescription, { htmlTag: "p" }, () => "dialog_description")
        ])
      )
    ];
  }
});

describe("dialog", () => {
  afterEach(() => {
    document.body.innerHTML = '<div id="app"></div>';
  });

  it("should be open by default", () => {
    mount(dialog, {
      props: { openByDefault: true },
      attachTo: document.getElementById("app") ?? document.body
    });

    expect(document.body.innerText).toMatchSnapshot();
  });

  it("should be closed by default", () => {
    mount(dialog, {
      props: { openByDefault: false },
      attachTo: document.getElementById("app") ?? document.body
    });
    expect(document.body.innerText).toMatchSnapshot();
  });

  it("should open", async () => {
    const wrapper = mount(dialog, { attachTo: document.getElementById("app") ?? document.body });

    wrapper.find("button").trigger("click");

    await nextTick();

    expect(document.body.innerText).toMatchSnapshot();
  });

  it("should close", async () => {
    const wrapper = mount(dialog, {
      props: { openByDefault: true },
      attachTo: document.getElementById("app") ?? document.body
    });

    wrapper.find("button").trigger("click");

    await nextTick();

    expect(document.body.innerText).toMatchSnapshot();
  });

  it("should trap the focus", async () => {
    //
  });

  it("should have aria description and title", async () => {
    //
  });
});
