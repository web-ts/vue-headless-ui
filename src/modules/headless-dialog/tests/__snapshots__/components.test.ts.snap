// Vitest Snapshot v1

exports[`dialog > should be closed by default 1`] = `"Open Dialog"`;

exports[`dialog > should be open by default 1`] = `"Open Dialogdialog_titledialog_description"`;

exports[`dialog > should close 1`] = `"Open Dialog"`;

exports[`dialog > should open 1`] = `"Open Dialogdialog_titledialog_description"`;
